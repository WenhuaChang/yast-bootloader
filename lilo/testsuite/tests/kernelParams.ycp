/**
 * File:
 *  kernelParams.ycp
 *
 * Module:
 *  Bootloader configurator
 *
 * Summary:
 *  Universal finctions for accessing kernel parameters.
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    include "testsuite.ycp";
    import "Pkg";
    map IREAD = $["target" : $[
            "string" : "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit",
            "tmpdir" : "/tmp"
        ], "etc" : $[
            "fstab" : [],
            "install_inf" : $[
                "Locale" : "en_EN",
                "InstMode" : "cd"
            ],
            "mtab" : []
        ], "probe" : $[
            "disk" : [],
            "boot_arch" : "grub",
            "architecture" : "i386",
            "has_apm" : true,
            "has_pcmcia" : false,
            "has_smp" : false,
            "system" : [],
            "floppy" : $[
                "manual" : []
            ], "cdrom" : $[
                "manual" : []
            ]
        ], "sysconfig" : $[
            "console" : $[
                "CONSOLE_FONT" : "lat2-16.psfu",
                "CONSOLE_SCREENMAP" : "trivial",
                "CONSOLE_UNICODEMAP" : "",
                "CONSOLE_MAGIC" : "(K",
                "CONSOLE_ENCODING" : "ISO-8859-2"
            ],
            "bootloader" : $[
                "LOADER_TYPE" : "grub"
            ]
        ], "md" : $[
            "config" : $[]
        ], "lvm" : $[
            "vg" : $[],
            "pv" : $[]
        ], "proc" : $[
            "swaps" : [],
            "mounts" : []
        ]
    ];
    TESTSUITE_INIT ([IREAD, $[], $[]], 0);
    include "lilo/misc.ycp";

    map READ = $["grub": $["sections" : $["linux" : $["kernel" : "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"]]],
		 "lilo": $["sections" : $["linux" : $["kernel" : "/boot/vmlinuz.suse", "vga" : "333", "root" : "/dev/hda3", "append" : "noarg ide=nodma init=/bin/bash"]]]];

    global define boot () ``{return "grub";}
    Lilo::mode = `grub;

    TEST (``(getKernelParam ("linux", "root")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "vga")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "init")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "noarg")), [READ, $[], $[]], 0);
    DUMP("==================================================================");
    TEST (``(setKernelParam ("linux", "root", "/dev/hda5")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "vga", "555")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "ide", "false")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "extra", "true")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "vga", "false")), [READ, $[], $[]], 0);
    DUMP("==================================================================");
    TEST (``(saveBootloaderConfig (true)), [READ, $[], $[]], 0);
    TEST (``(saveBootloaderConfig (false)), [READ, $[], $[]], 0);

    global define boot () ``{return "lilo";}
    Lilo::mode = `lilo;

    DUMP("==================================================================");
    TEST (``(getKernelParam ("linux", "root")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "vga")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "init")), [READ, $[], $[]], 0);
    TEST (``(getKernelParam ("linux", "noarg")), [READ, $[], $[]], 0);
    DUMP("==================================================================");
    TEST (``(setKernelParam ("linux", "root", "/dev/hda5")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "vga", "555")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "ide", "false")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "extra", "true")), [READ, $[], $[]], 0);
    TEST (``(setKernelParam ("linux", "vga", "true")), [READ, $[], $[]], 0);
    DUMP("==================================================================");
    TEST (``(saveBootloaderConfig (true)), [READ, $[], $[]], 0);
    TEST (``(saveBootloaderConfig (false)), [READ, $[], $[]], 0);

}

