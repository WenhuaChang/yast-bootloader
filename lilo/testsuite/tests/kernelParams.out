Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.tmpdir "/tmp"
Read	.target.size "/etc/install.inf" 0
Read	.etc.install_inf.Sourcemounted 0
Read	.etc.install_inf.AutoYaST 0
Read	.etc.install_inf.Demo 0
Read	.etc.install_inf.Textmode 0
Read	.etc.install_inf.Manual 0
Read	.etc.install_inf.Console 0
Read	.etc.install_inf.Braille 0
Read	.etc.install_inf.VNC 0
Read	.etc.install_inf.UseSSH 0
Read	.etc.install_inf.Locale "en_EN"
Read	.etc.install_inf.InstMode "cd"
Read	.sysconfig.console.CONSOLE_FONT "lat2-16.psfu"
Read	.sysconfig.console.CONSOLE_SCREENMAP "trivial"
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC "(K"
Read	.sysconfig.console.CONSOLE_ENCODING "ISO-8859-2"
Execute	.target.bash "/usr/bin/setfont" 0
Read	.probe.floppy.manual []
Read	.probe.cdrom.manual []
Read	.probe.boot_arch "grub"
Read	.probe.disk []
Read	.md.config $[]
Read	.lvm.vg $[]
Read	.lvm.pv $[]
Read	.target.size "/etc/cryptotab" 0
Read	.proc.swaps []
Write	.dumpto.tmp.targetMap_i $[] true
Read	.proc.mounts []
Read	.proc.swaps []
Read	.etc.mtab []
Read	.target.size "/etc/fstab" 0
Read	.target.size "/etc/cryptotab" 0
Read	.etc.fstab []
Read	.target.string "/etc/grub.conf" "root (hd0,0)\ninstall /grub/stage1 d (hd0) /grub/stage2 0x8000 (hd0,0)/grub/menu.lst\nquit"
Read	.probe.disk []
Read	.probe.boot_arch "grub"
Read	.target.size "/etc/sysconfig/bootloader" 0
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Return	/dev/hda3
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Return	333
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Return	/bin/bash
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Return	true
Dump	==================================================================
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Write	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda5 vga=333 ide=nodma noarg init=/bin/bash" true
Write	."grub" nil true
Return	true
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Write	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=555 ide=nodma noarg init=/bin/bash" true
Write	."grub" nil true
Return	true
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Write	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 noarg init=/bin/bash" true
Write	."grub" nil true
Return	true
Read	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash"
Write	."grub"."sections"."linux"."kernel" "(hd0,0)/vmlinuz.suse noval root=/dev/hda3 vga=333 ide=nodma noarg init=/bin/bash extra" true
Write	."grub" nil true
Return	true
Return	false
Dump	==================================================================
Write	."grub" nil true
Execute	.target.bash "/sbin/mk_initrd" 0
Return	true
Write	."grub" nil true
Return	true
Dump	==================================================================
Read	."lilo"."sections"."linux"."root" "/dev/hda3"
Return	/dev/hda3
Read	."lilo"."sections"."linux"."vga" "333"
Return	333
Read	."lilo"."sections"."linux"."append" "noarg ide=nodma init=/bin/bash"
Return	/bin/bash
Read	."lilo"."sections"."linux"."append" "noarg ide=nodma init=/bin/bash"
Return	true
Dump	==================================================================
Write	."lilo"."sections"."linux"."root" "/dev/hda5" true
Write	."lilo" nil true
Return	true
Write	."lilo"."sections"."linux"."vga" "555" true
Write	."lilo" nil true
Return	true
Read	."lilo"."sections"."linux"."append" "noarg ide=nodma init=/bin/bash"
Write	."lilo"."sections"."linux"."append" "noarg init=/bin/bash" true
Write	."lilo" nil true
Return	true
Read	."lilo"."sections"."linux"."append" "noarg ide=nodma init=/bin/bash"
Write	."lilo"."sections"."linux"."append" "noarg ide=nodma init=/bin/bash extra" true
Write	."lilo" nil true
Return	true
Return	false
Dump	==================================================================
Write	."lilo" nil true
Execute	.target.bash "/sbin/mk_initrd" 0
Execute	.target.bash "lilo" $[] 0
Return	true
Write	."lilo" nil true
Execute	.target.bash "lilo" $[] 0
Return	true
